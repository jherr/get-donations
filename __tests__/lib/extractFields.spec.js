const fs = require("fs");
const path = require("path");
const { DOMParser } = require("xmldom");

const {
  isValidFormat,
  extractFields,
  getTaxYear,
} = require("../../lib/extractFields");

describe("extractFields", () => {
  let penn2018 = null;
  let northwestern2018 = null;

  beforeAll(() => {
    penn2018 = new DOMParser().parseFromString(
      fs.readFileSync(
        path.resolve(__dirname, "./__fixtures__/penn-2018.xml"),
        "utf8"
      ),
      "text/xml"
    );
    northwestern2018 = new DOMParser().parseFromString(
      fs.readFileSync(
        path.resolve(__dirname, "./__fixtures__/northwestern-2018.xml"),
        "utf8"
      ),
      "text/xml"
    );
  });

  it("should extract the right tax year", () => {
    expect(getTaxYear(penn2018)).toBe(2018);
  });

  it("should say that the XML file is valid", () => {
    expect(isValidFormat(penn2018)).toBe(true);
  });

  it("penn-2018", () => {
    expect(extractFields(penn2018)).toMatchSnapshot();
  });

  it("northwestern-2018", () => {
    expect(extractFields(northwestern2018)).toMatchSnapshot();
  });
});
