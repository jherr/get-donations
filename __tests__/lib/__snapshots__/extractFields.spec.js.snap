// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractFields northwestern-2018 1`] = `
Object {
  "Sch990_IX_25_A": 6150079387,
  "Sch990_IX_7_A": 2053808142,
  "SchH_III_2": 59785393,
  "SchH_III_3": "",
  "SchH_III_5": 1369989829,
  "SchH_III_6": 2019187038,
  "SchH_III_7": -649197209,
  "SchH_II_10_a": 0,
  "SchH_II_10_b": 0,
  "SchH_II_10_c": 3236218,
  "SchH_II_10_d": 0,
  "SchH_II_10_e": 3236218,
  "SchH_II_10_f": 0.0005,
  "SchH_II_1_a": "",
  "SchH_II_1_b": "",
  "SchH_II_1_c": "",
  "SchH_II_1_d": "",
  "SchH_II_1_e": 0,
  "SchH_II_1_f": 0,
  "SchH_II_2_a": "",
  "SchH_II_2_b": "",
  "SchH_II_2_c": "",
  "SchH_II_2_d": "",
  "SchH_II_2_e": 0,
  "SchH_II_2_f": 0,
  "SchH_II_3_a": "",
  "SchH_II_3_b": "",
  "SchH_II_3_c": "",
  "SchH_II_3_d": "",
  "SchH_II_3_e": 0,
  "SchH_II_3_f": 0,
  "SchH_II_4_a": "",
  "SchH_II_4_b": "",
  "SchH_II_4_c": "",
  "SchH_II_4_d": "",
  "SchH_II_4_e": 0,
  "SchH_II_4_f": 0,
  "SchH_II_5_a": "",
  "SchH_II_5_b": "",
  "SchH_II_5_c": "",
  "SchH_II_5_d": "",
  "SchH_II_5_e": 0,
  "SchH_II_5_f": 0,
  "SchH_II_6_a": "",
  "SchH_II_6_b": "",
  "SchH_II_6_c": "",
  "SchH_II_6_d": "",
  "SchH_II_6_e": 0,
  "SchH_II_6_f": 0,
  "SchH_II_7_a": "",
  "SchH_II_7_b": "",
  "SchH_II_7_c": "",
  "SchH_II_7_d": "",
  "SchH_II_7_e": 0,
  "SchH_II_7_f": 0,
  "SchH_II_8_a": "",
  "SchH_II_8_b": "",
  "SchH_II_8_c": 3236218,
  "SchH_II_8_d": "",
  "SchH_II_8_e": 3236218,
  "SchH_II_8_f": 0.0005,
  "SchH_II_9_a": "",
  "SchH_II_9_b": "",
  "SchH_II_9_c": "",
  "SchH_II_9_d": "",
  "SchH_II_9_e": 0,
  "SchH_II_9_f": 0,
  "SchH_I_7_a_c": 72573568,
  "SchH_I_7_a_d": 4238622,
  "SchH_I_7_a_e": 68334946,
  "SchH_I_7_a_f": 0.0116,
  "SchH_I_7_b_c": 556716298,
  "SchH_I_7_b_d": 403812719,
  "SchH_I_7_b_e": 152903579,
  "SchH_I_7_b_f": 0.026,
  "SchH_I_7_d_c": 629289866,
  "SchH_I_7_d_d": 408051341,
  "SchH_I_7_d_e": 221238525,
  "SchH_I_7_d_f": 0.0376,
  "SchH_I_7_e_c": 5268251,
  "SchH_I_7_e_d": "",
  "SchH_I_7_e_e": 5268251,
  "SchH_I_7_e_f": 0.0009,
  "SchH_I_7_f_c": 59167843,
  "SchH_I_7_f_d": 15642449,
  "SchH_I_7_f_e": 43525394,
  "SchH_I_7_f_f": 0.0074,
  "SchH_I_7_g_c": 21136520,
  "SchH_I_7_g_d": "",
  "SchH_I_7_g_e": 21136520,
  "SchH_I_7_g_f": 0.0036,
  "SchH_I_7_h_c": 51876527,
  "SchH_I_7_h_d": "",
  "SchH_I_7_h_e": 51876527,
  "SchH_I_7_h_f": 0.0088,
  "SchH_I_7_i_c": 2553556,
  "SchH_I_7_i_d": "",
  "SchH_I_7_i_e": 2553556,
  "SchH_I_7_i_f": 0.0004,
  "SchH_I_7_j_c": 140002697,
  "SchH_I_7_j_d": 15642449,
  "SchH_I_7_j_e": 124360248,
  "SchH_I_7_j_f": 0.0211,
  "SchH_I_7_k_c": 769292563,
  "SchH_I_7_k_d": 423693790,
  "SchH_I_7_k_e": 345598773,
  "SchH_I_7_k_f": 0.0587,
  "businessName": "Northwestern Memorial HealthCare Group",
  "ein": "364724966",
  "preparerName": "ERNST & YOUNG US LLP",
  "returnTimeStamp": "2020-07-15T11:05:22-05:00",
  "state": "IL",
  "taxEndDate": "2019-08-31",
  "taxYear": 2018,
  "zipCode": "606112908",
}
`;

exports[`extractFields penn-2018 1`] = `
Object {
  "Sch990_IX_25_A": 6978995000,
  "Sch990_IX_7_A": 3029877958,
  "SchH_III_2": 75526511,
  "SchH_III_3": 27994066,
  "SchH_III_5": 617952755,
  "SchH_III_6": 665359239,
  "SchH_III_7": -47406484,
  "SchH_II_10_a": "",
  "SchH_II_10_b": "",
  "SchH_II_10_c": "",
  "SchH_II_10_d": "",
  "SchH_II_10_e": "",
  "SchH_II_10_f": "",
  "SchH_II_1_a": "",
  "SchH_II_1_b": "",
  "SchH_II_1_c": "",
  "SchH_II_1_d": "",
  "SchH_II_1_e": "",
  "SchH_II_1_f": "",
  "SchH_II_2_a": "",
  "SchH_II_2_b": "",
  "SchH_II_2_c": "",
  "SchH_II_2_d": "",
  "SchH_II_2_e": "",
  "SchH_II_2_f": "",
  "SchH_II_3_a": "",
  "SchH_II_3_b": "",
  "SchH_II_3_c": "",
  "SchH_II_3_d": "",
  "SchH_II_3_e": "",
  "SchH_II_3_f": "",
  "SchH_II_4_a": "",
  "SchH_II_4_b": "",
  "SchH_II_4_c": "",
  "SchH_II_4_d": "",
  "SchH_II_4_e": "",
  "SchH_II_4_f": "",
  "SchH_II_5_a": "",
  "SchH_II_5_b": "",
  "SchH_II_5_c": "",
  "SchH_II_5_d": "",
  "SchH_II_5_e": "",
  "SchH_II_5_f": "",
  "SchH_II_6_a": "",
  "SchH_II_6_b": "",
  "SchH_II_6_c": "",
  "SchH_II_6_d": "",
  "SchH_II_6_e": "",
  "SchH_II_6_f": "",
  "SchH_II_7_a": "",
  "SchH_II_7_b": "",
  "SchH_II_7_c": "",
  "SchH_II_7_d": "",
  "SchH_II_7_e": "",
  "SchH_II_7_f": "",
  "SchH_II_8_a": "",
  "SchH_II_8_b": "",
  "SchH_II_8_c": "",
  "SchH_II_8_d": "",
  "SchH_II_8_e": "",
  "SchH_II_8_f": "",
  "SchH_II_9_a": "",
  "SchH_II_9_b": "",
  "SchH_II_9_c": "",
  "SchH_II_9_d": "",
  "SchH_II_9_e": "",
  "SchH_II_9_f": "",
  "SchH_I_7_a_c": 4167050,
  "SchH_I_7_a_d": 0,
  "SchH_I_7_a_e": 4167050,
  "SchH_I_7_a_f": 0.0006,
  "SchH_I_7_b_c": 401988953,
  "SchH_I_7_b_d": 299466442,
  "SchH_I_7_b_e": 102522511,
  "SchH_I_7_b_f": 0.0147,
  "SchH_I_7_d_c": 406156003,
  "SchH_I_7_d_d": 299466442,
  "SchH_I_7_d_e": 106689561,
  "SchH_I_7_d_f": 0.0153,
  "SchH_I_7_e_c": "",
  "SchH_I_7_e_d": "",
  "SchH_I_7_e_e": "",
  "SchH_I_7_e_f": "",
  "SchH_I_7_f_c": 138767882,
  "SchH_I_7_f_d": 55884226,
  "SchH_I_7_f_e": 82883656,
  "SchH_I_7_f_f": 0.0119,
  "SchH_I_7_g_c": "",
  "SchH_I_7_g_d": "",
  "SchH_I_7_g_e": "",
  "SchH_I_7_g_f": "",
  "SchH_I_7_h_c": 742905052,
  "SchH_I_7_h_d": 603889647,
  "SchH_I_7_h_e": 139015405,
  "SchH_I_7_h_f": 0.0199,
  "SchH_I_7_i_c": "",
  "SchH_I_7_i_d": "",
  "SchH_I_7_i_e": "",
  "SchH_I_7_i_f": "",
  "SchH_I_7_j_c": 881672934,
  "SchH_I_7_j_d": 659773873,
  "SchH_I_7_j_e": 221899061,
  "SchH_I_7_j_f": 0.0318,
  "SchH_I_7_k_c": 1287828937,
  "SchH_I_7_k_d": 959240315,
  "SchH_I_7_k_e": 328588622,
  "SchH_I_7_k_f": 0.0471,
  "businessName": "TRUSTEES OF THE UNIVERSITY OF PENNSYLVANIA",
  "ein": "231352685",
  "preparerName": "PricewaterhouseCoopers LLP",
  "returnTimeStamp": "2020-07-10T15:52:51-05:00",
  "state": "PA",
  "taxEndDate": "2019-06-30",
  "taxYear": 2018,
  "zipCode": "19104",
}
`;
